joomla/joomla-cms
# [4.0] Repackage the js of fancy select field

Pull Request for Issue # .

### Summary of Changes
- Remove the choices.js from the vendor folder
- Use `import` in the fancy select field

### Why?
Well, custom elements (and Web Components) are better if they don't depend on other scripts, eg append the HTML markup and the script element should be the only step for using them.

Also, personally, I'm very worried that the library is already abandoned and shipping it in the vendor folder the project creates the same situation that existed with chosen. Let's not do the same mistake. By using import to utilize choices we are encapsulating the whole thing ONLY in the fancy select script source and thus someone could swap the library or completely rewrite the custom element without any dependency. In short, it's more future proof this way...


### Testing Instructions
Apply that patch (needs `npm ci`
Check as many fancy select (the one with the search) in as many edit forms as possible
![Screenshot 2021-05-11 at 13 54 15](https://user-images.githubusercontent.com/3889375/117811069-6ec5e680-b260-11eb-947e-26f482fb5056.png)


### Actual result BEFORE applying this Pull Request
Works


### Expected result AFTER applying this Pull Request
Works, 1 HTTP request less (but this PR is not about reducing the HTTP requests, HTTP3 is already here and the number of requests is completely irrelevant for performance)


### Documentation Changes Required

No

@Fedik are you ok with this one?
